### ISSUE TITLE
question about attention_weight compute method

### ISSUE THREAD

**Issue by Johnson-yue** [garibida/cross-image-attention#12]
Hi, thanks for your sharing , I have a question about attention_weight compute method
In your code :

https://github.com/garibida/cross-image-attention/blob/a624d05bc039fe3f204dd9d1ffe80798b5669bfb/utils/attention_utils.py#L24C58-L24C58

`attn_weight = softmax( k @ q.T / q.shape[-1] ** 0.5 )`

but Normal Implement : 
`attn_weight = softmax( k @ q.T / (q.shape[-1] / attn_head) ** 0.5 )`

why are you doing this ?  I`m so confuse  about this

**Comment by yuval-alaluf** [comment]
Hi, we are using the standard scaled dot product attention as implemented in pytorch: https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html

I assume that there are many variations of the dot product with minor differences between them, but there is no specific reason we implemented it as we did. I do not believe that you will see any differences if you add the normalization of the number of heads.

**Comment by Johnson-yue** [comment]
ok， thank your reply ， performance are  good for two compute method
